### ISSUE TITLE
DataBufferObservable should propagate errors from the Buffer

### ISSUE THREAD

**Issue by edenman** [mcharmas/Android-ReactiveLocation#147]
If the Buffer object also `implements Result`, it'd be rad to have non-success status (as determined by `getStatus().isSuccess()`) propagated to `onError`.

https://developers.google.com/android/reference/com/google/android/gms/common/api/Result

Let me know if this is something you'd be interested in, and I can whip up a PR.

**Comment by mcharmas** [comment]
Good idea. I must have missed that. If you are still interested please preapre PR.
